app-creator/client/src/utils/helpers.ts

```diff
@@ -163,3 +165,47 @@ export function setUrlParam(key: string, value: string): URL {
 
   return url;
 }
+
+/**
+ * Adds background color to an element. Used when adding elements to the scratch panel.
+ */
+const WidgetsRequiringBackground = new Set([
+  WidgetType.LABEL,
+  WidgetType.CHECKBOX,
+  WidgetType.SLIDER,
+]);
+
+export function addBackgroundColorToSharedWidget(element: HTMLElement) {
+  const elementStyle: { [key: string]: string } = Object.assign(
+    {},
+    store.getState().template.widgets[element.id].style
+  );
+
+  const type = getWidgetType(element.id);
+  if (
+    elementStyle.color === '#ffffff' &&
+    WidgetsRequiringBackground.has(type as WidgetType)
+  ) {
+    const paletteBackground = store.getState().selectedPalette.backgroundColor;
+    elementStyle.backgroundColor =
+      paletteBackground === '#ffffff' ? '#000000' : paletteBackground;
+    elementStyle.borderRadius = '8px';
+    elementStyle.padding = '8px';
+    (element as EEWidget).setStyle(elementStyle);
+  }
+}
+
+/**
+ * Removes background color from an element. Used when adding elements to the scratch panel.
+ */
+export function removeBackgroundColorFromSharedWidget(element: HTMLElement) {
+  const elementStyle: { [key: string]: string } = Object.assign(
+    {},
+    store.getState().template.widgets[element.id].style
+  );
+
+  elementStyle.backgroundColor = elementStyle.backgroundColor;
+  elementStyle.borderRadius = elementStyle.borderRadius ?? '4px';
+  elementStyle.padding = elementStyle.padding ?? '0px';
```

Comment:
Retrieve these defaults from where they are defined, if possible.

Reply:
I have referenced the default padding value. `borderRadius` however is not part of shared styles so I have kept it as is.